back/courses/views.py

```diff
@@ -126,3 +139,27 @@ def update_timeslots(request):
             context["date_updated"] = date
 
     return render(request, "courses/update_timeslots.html", context)
+
+
+@login_required
+def join_group(request, group_id, action):
+    group = get_object_or_404(Group, id=group_id)
+    student = get_object_or_404(Student, user__id=request.user.id)
+    for group_to_delete in group.course.groups.all():
+        student.course.remove(group_to_delete)
+    if action == "Join_group" or action == "Join_course":
+        enrolment = Enrolment(
+            group=group,
+            student=student,
+        )
+        enrolment.save()
+        if action == "Join_group":
+            for group in group.course.groups.all().filter(number__isnull=True):
```

Comment:
I think a comment to briefly explains why we do the loop would be a good idea

Reply:
Got it

Reply:
I erased the for loop and replaced it with something less ambiguous